.github/workflows/ci-workflow.yml

```diff
@@ -1,32 +1,45 @@
+---
+
 name: Compilation & tests
 
 on:
   push:
-    branches: [ "master", "dev" ]
+    branches: ["master", "dev", "main", "develop"]
   pull_request:
-    branches: [ "master", "dev" ]
+    branches: ["master", "dev", "main", "develop"]
 
 jobs:
   job_build_debug:
     name: Build debug
+    strategy:
+      matrix:
+        device: [nanos, nanox, nanosp]
     runs-on: ubuntu-latest
 
     container:
       image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
 
     steps:
       - name: Clone
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Build
         run: |
-          make DEBUG=1
+          echo "Building for ${{ matrix.device }}..." && \
+          eval "BOLOS_SDK=\$$(echo ${{ matrix.device }} | tr [:lower:] [:upper:])_SDK" && \
+          echo "BOLOS_SDK value will be ${BOLOS_SDK}" && \
+          make BOLOS_SDK=${BOLOS_SDK} DEBUG=1
+
+      - name: Prepare upload
+        run: |
+          find build/ -mindepth 2 -maxdepth 2 -type d ! -name 'bin' -exec rm -r {} +
 
       - name: Upload app binary
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v4
         with:
-          name: hathor-app-debug
-          path: bin
+          name: hathor-app-debug-${{ matrix.device }}
+          path: build/
+          if-no-files-found: error
```

Comment:
We need to change the artifact name since we will upload one for each device there would be a conflict.